### ISSUE TITLE
Existing warning results in error

### ISSUE THREAD

**Issue by sergePSE** [litespeedtech/lsquic-client#23]
On make:
lsquic-client/src/liblsquic/lsquic_logger.h:129:12: error: this statement may fall through [-Werror=implicit-fallthrough=]
         if (LSQ_LOG_ENABLED(level))                                          \
            ^

cmake 3.10.2
make 4.1
gcc 7.3.0

**Comment by dtikhonov** [comment]
Thank you for the report.  Would you mind pasting a few more lines of the compiler error output, please?

**Comment by sergePSE** [comment]
My full output.

[  1%] Building C object src/liblsquic/CMakeFiles/lsquic.dir/lsquic_chsk_stream.c.o
In file included from /home/sergei/Documents/git/lsquic-client/src/liblsquic/lsquic_chsk_stream.c:27:0:
/home/sergei/Documents/git/lsquic-client/src/liblsquic/lsquic_chsk_stream.c: In function ‘hsk_client_on_read’:
/home/sergei/Documents/git/lsquic-client/src/liblsquic/lsquic_logger.h:129:12: error: this statement may fall through [-Werror=implicit-fallthrough=]
         if (LSQ_LOG_ENABLED(level))                                          \
            ^
/home/sergei/Documents/git/lsquic-client/src/liblsquic/lsquic_logger.h:162:24: note: in expansion of macro ‘LSQ_LOG2’
 #       define LSQ_LOG LSQ_LOG2
                        ^~~~~~~~
/home/sergei/Documents/git/lsquic-client/src/liblsquic/lsquic_logger.h:173:26: note: in expansion of macro ‘LSQ_LOG’
 #define LSQ_WARN(...)    LSQ_LOG(LSQ_LOG_WARN,   __VA_ARGS__)
                          ^~~~~~~
/home/sergei/Documents/git/lsquic-client/src/liblsquic/lsquic_chsk_stream.c:119:9: note: in expansion of macro ‘LSQ_WARN’
         LSQ_WARN("lsquic_enc_session_handle_chlo_reply returned unknown value %d", s);
         ^~~~~~~~
/home/sergei/Documents/git/lsquic-client/src/liblsquic/lsquic_chsk_stream.c:120:5: note: here
     case DATA_FORMAT_ERROR:
     ^~~~
cc1: all warnings being treated as errors
src/liblsquic/CMakeFiles/lsquic.dir/build.make:134: recipe for target 'src/liblsquic/CMakeFiles/lsquic.dir/lsquic_chsk_stream.c.o' failed
make[2]: *** [src/liblsquic/CMakeFiles/lsquic.dir/lsquic_chsk_stream.c.o] Error 1
CMakeFiles/Makefile2:172: recipe for target 'src/liblsquic/CMakeFiles/lsquic.dir/all' failed
make[1]: *** [src/liblsquic/CMakeFiles/lsquic.dir/all] Error 2
Makefile:94: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by dtikhonov** [comment]
I see, it's this code:

```
    default:
        LSQ_WARN("lsquic_enc_session_handle_chlo_reply returned unknown value %d", s);
    case DATA_FORMAT_ERROR:
        LSQ_INFO("lsquic_enc_session_handle_chlo_reply returned an error");
```

We _want_ it to fall through.  I wonder what the new gcc wants from us here? 🤔

**Comment by dtikhonov** [comment]
Bing finds this:

  https://developers.redhat.com/blog/2017/03/10/wimplicit-fallthrough-in-gcc-7/

Boy, talk about hand-holding...

**Comment by sergePSE** [comment]
Thank you for your response, I've disabled a flag -Werror, but you had a comment, that you want prevent warnings. As I understand we should add explicit comments.

**Comment by litespeedtech** [comment]
These warnings have been addressed: see pull request #26.
